src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/chatpageobject/ChatPage.java

```diff
@@ -0,0 +1,408 @@
+  private static final String USER_UNBAN_CONFIRM_MESSAGE =
+          "//div[@class='Chat']//li[contains(text(), 'has ended the Chat ban for %s')]";
+  private static final String USER_SELECTOR = "#user-%s";
+  private static final String PRIVATE_MESSAGE_USER_SELECTOR = "#priv-user-%s";
+  private static final String PATH_MESSAGE_ON_CHAT = "//span[@class='message'][contains(text(), '%s')]";
+  private static final String NOTIFICATION_COUNTER =
+          "//span[@class='splotch' and contains(text(), '%s')]";
+
+  private static final int REGULAR_USER_DROPDOWN_ELEMENTS_COUNT = 3;
+
+  public ChatPage open() {
+    getUrl(urlBuilder.getUrlForWiki() + URLsContent.SPECIAL_CHAT);
+
+    return this;
+  }
+
+  private WebElement userOnChat(String userName, String message) {
+    return driver.findElement(By.cssSelector(String.format(message, userName)));
+  }
+
+  private WebElement userPostedMessage(String message) {
+    return driver.findElement(By.xpath(String.format(PATH_MESSAGE_ON_CHAT, message)));
+  }
+
+  private WebElement userPrivateMessageNotification(int notificationCount) {
+    return driver.findElement(By.xpath(String.format(NOTIFICATION_COUNTER, notificationCount)));
+  }
+
+  private WebElement getUserUnbanMessage(String userName) {
+    return driver.findElement(By.xpath(String.format(USER_UNBAN_CONFIRM_MESSAGE, userName)));
+  }
+
+  private WebElement getUserUnbanLink(String userName) {
+   return driver.findElement(By.xpath(String.format(USER_UNBAN_LINK, userName)));
+  }
+
+  public boolean isUserOnChat() {
+    try {
+      wait.forElementVisible(chatInlineAlert);
+      return true;
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public boolean isMessageOnChat(String message) {
+    try {
+      WebElement userPostedMessage = userPostedMessage(message);
+      wait.forElementVisible(userPostedMessage);
+      return true;
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public boolean isPrivateMessageHeaderDispayed() {
+    try {
+      wait.forElementVisible(privateMessagesHeader);
+      return true;
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public boolean isPermissionsErrorTitleDisplayed() {
+    try {
+      wait.forElementVisible(permissionsErrorTitle);
+      return true;
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public boolean isPrivateMessageNotificationDisplayed() {
+    try {
+      wait.forElementVisible(privateMessageNotification);
+      return true;
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public boolean isPrivateNotificationCountDisplayed(int notificationCount) {
+    try {
+      wait.forElementVisible(userPrivateMessageNotification(notificationCount));
+      return true;
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public boolean isPrivateMessageButtonDisplayed() {
+    try {
+      return privateMassageButton.isDisplayed();
+    } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)